### PR TITLE
feat: Add SettlementSeriesVersion to AggregatedTimeSeriesRequest

### PR DESCRIPTION
**Pull request by ebbeknudsen**
## Description
Add SettlementSeriesVersion to AggregatedTimeSeriesRequest
- Validate valid data for SettlementSeriesVersion (and BusinessReason)
- Update SQL request to account for SettlementSeriesVersion

## References
Task: https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/16

### CODE REVIEW

File: source/dotnet/wholesale-api/Edi.UnitTests/Validators/AggregatedTimeSeriesRequestValidatorTests.cs [hunk 1]
```diff
@@ -126,29 +114,41 @@ public void Validate_WhenSettlementMethodIsInvalid_ReturnsUnsuccessfulValidation
...
 
         // Assert
-        validationErrors.Should().ContainSingle();
+        validationErrors.Should().ContainSingle()
+            .Which.ErrorCode.Should().Be("D15");
     }
 
     [Fact]
-    public void Validate_AsEnergySupplierTotalConsumption_ReturnsUnsuccessfulValidation()
+    public void Validate_WhenSettlementMethodSeriesIsInvalid_ReturnsUnsuccessfulValidation()
```
Comment (RasmusGraabaek) [inline source/dotnet/wholesale-api/Edi.UnitTests/Validators/AggregatedTimeSeriesRequestValidatorTests.cs:129]:
```suggestion
    public void Validate_WhenSettlementSeriesVersionIsInvalid_ReturnsUnsuccessfulValidation()
```

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/RequestCalculationResult/RequestCalculationResultQueriesTests.cs [hunk 2]
```diff
@@ -281,30 +290,176 @@ public RequestCalculationResultQueriesTests(DatabricksSqlStatementApiFixture fix
...
+        actual.PeriodStart.Should().Be(startOfPeriodFilter);
+        actual.PeriodEnd.Should().Be(endOfPeriodFilter);
+        actual.ProcessType.Should().Be(processTypeFilter);
+        actual.TimeSeriesType.Should().Be(timeSeriesTypeFilter);
+        actual.TimeSeriesPoints
+            .Select(p => p.Quantity.ToString(CultureInfo.InvariantCulture))
+            .ToArray()
+            .Should()
+            .Equal(FirstQuantitySecondCorrection, SecondQuantitySecondCorrection)
+            ;
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/RequestCalculationResult/RequestCalculationResultQueriesTests.cs:383]: :D?

**Review by RasmusGraabaek — DISMISSED** [review]

**Review by lasrinnil — APPROVED** [review]
LGTM

File: source/dotnet/wholesale-api/Edi.UnitTests/Validators/SettlementSeriesVersionValidatorTest.cs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+using Energinet.DataHub.Wholesale.EDI.Validation;
+using Energinet.DataHub.Wholesale.EDI.Validation.AggregatedTimeSerie.Rules;
+using FluentAssertions;
+using Xunit;
+
+namespace Energinet.DataHub.Wholesale.EDI.UnitTests.Validators;
+
+public class SettlementSeriesVersionValidatorTest
+{
+    private static readonly ValidationError _expectedInvalidSettlementMethodError = new("SettlementSeriesVersion kan kun benyttes i kombination med D32 og skal være enten D01, D02 eller D03 / SettlementSeriesVersion can only be used in combination with D32 and must be either D01, D02 or D03", "E86");
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/Edi.UnitTests/Validators/SettlementSeriesVersionValidatorTest.cs:26]: Danish?

File: source/dotnet/wholesale-api/Edi/Validation/AggregatedTimeSerie/Rules/SettlementSeriesVersionValidationRule.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    public IList<ValidationError> Validate(AggregatedTimeSeriesRequest subject)
+    {
+        if (!subject.HasSettlementSeriesVersion)
+            return NoError;
+
+        if (subject.BusinessReason != BusinessReason.Correction)
+            return InvalidSettlementMethodError;
+
+        if (!_validSettlementSeriesVersions.Contains(subject.SettlementSeriesVersion))
+            return InvalidSettlementMethodError;
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/Edi/Validation/AggregatedTimeSerie/Rules/SettlementSeriesVersionValidationRule.cs:40]: The business doesn't desire more specific error messages in each case?

Comment (ebbeknudsen) [reply source/dotnet/wholesale-api/Edi/Validation/AggregatedTimeSerie/Rules/SettlementSeriesVersionValidationRule.cs:40]: Sadly, no. We have to follow the validation errors/cases from the old system :(

File: source/dotnet/wholesale-api/Edi.UnitTests/Validators/SettlementSeriesVersionValidatorTest.cs (hunk 3 above)
Comment (ebbeknudsen) [reply source/dotnet/wholesale-api/Edi.UnitTests/Validators/SettlementSeriesVersionValidatorTest.cs:26]: @BjarkeMeier There's a seperate task on our backlog for removing the domain codes: https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/46

It's not been prioritized yet, but hopefully we'll be able to implement it soon.

**Review by johevemi — APPROVED** [review]

**Pull request merged**